src/commands/NewCamelKameletCommand.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { QuickPickItem, Uri, commands, window } from "vscode";
+import { AbstractNewCamelRouteCommand } from "./AbstractNewCamelRouteCommand";
+import { CamelInitJBangTask } from "../tasks/CamelInitJBangTask";
+import { CamelRouteDSL } from "./AbstractCamelCommand";
+import path from "path";
+
+export class NewCamelKameletCommand extends AbstractNewCamelRouteCommand {
+
+	public static readonly ID_COMMAND_CAMEL_KAMELET_YAML = 'kaoto.camel.jbang.init.kamelet.yaml';
+	protected static readonly PROGRESS_NOTIFICATION_MESSAGE = 'Creating a new Kamelet file...';
+
+	private kameletType: string = '';
+
+	public async create(): Promise<void> {
+		const wsFolder = await this.showWorkspaceFolderPick();
```

Comment:
When escaping the workspace folder selection, instead of aborting the command, it goes to next step and ask for a folder to choos